### DISCUSSION TITLE
Integration with Libretranslate

### DISCUSSION THREAD

**Discussion by matgianesini** [WeblateOrg/weblate#11285] [Q&A]
Hello! I’m trying to connect my self hosted Weblate and Libretranslate instances, but when I add my Libretranslate URL in Weblate, I get this error: `* Could not fetch translation: 400 Client Error: BAD REQUEST for url: [MY LT INSTANCE’S URL]/translate`.

If I try the public Libretranslate instance, it accepts that without issue and I get automatic suggestions working. This leads me to believe the problem is my Libretranslate configuration. What’s strange is that other than connecting with Weblate, my Libretranslate instance seems to work the same as the public instance (I can make requests to its /translate endpoint without issue).

What I’ve tried:

- Setting up CORS to accept requests from my Weblate instance’s URL. That didn’t change anything.
- Configure all instances under the same docker-compose.yml, which I share below, in order to leverage Docker networking which would mitigate any errors related to network permissions. I couldn’t complete this test because Weblate won’t accept http://libretranslate:5000 as the Libretranslate URL by saying “Enter a valid URL”.
- As far as connecting using actual domain names, both the Weblate and the Libretranslate are using SSL and they’re accesible through `http://lang.mydomain.com` and `http://translate.mydomain.com` respectively. The SSL is given by the nginx reverse proxy that serves them.
- I’ve also tried explicitly setting Libretranslate’s SSL argument to `True` but that seems to do nothing.

If anyone wants to take a look around my instances, I can create an account for you if you give me a throwaway email address.

**Comment by nijel** [comment]
If you are using Weblate older than 5.4, consider upgrading, you might be hitting 
issue fixed by 48064f378969ba1dbd4762a89969349d12ad6645.

Besides that, I'd guess that it's a wrong/missing key. 068b6de9d31768a6099bc9a34c9721811042dc92 should give be a better insight what is wrong in the upcoming release.
